tests/testDetectAndMeasure.py

```diff
@@ -0,0 +1,134 @@
+from lsst.afw.geom import Box2I, Point2I, Extent2I
+from lsst.afw.table import SourceTable
+from lsst.daf.persistence import Butler
+from lsst.meas.algorithms.testUtils import plantSources
+from lsst.pipe.tasks.detectAndMeasure import DetectAndMeasureTask
+
+try:
+    obsTestDir = getPackageDir("obs_test")
+    InputDir = os.path.join(obsTestDir, "data", "input")
+except Exception:
+    InputDir = None
+
+class ProcessCcdTestCase(lsst.utils.tests.TestCase):
+    def testBasics(self):
+        """Test detection and measurement on simple synthesized data
+        """
+        bbox = Box2I(Point2I(256, 100), Extent2I(128, 127))
+        minCounts = 5000
+        maxCounts = 50000
+        starSigma = 1.5
+        numX = 5
+        numY = 5
+        coordList = self.makeCoordList(
+            bbox = bbox,
+            numX = numX,
+            numY = numY,
+            minCounts = minCounts,
+            maxCounts = maxCounts,
+            sigma = starSigma,
+        )
+        kwid = 11 # kernel width
+        sky = 2000
+        # create an exposure without a Wcs; add the Wcs later
+        exposure = plantSources(bbox=bbox, kwid=kwid, sky=sky, coordList=coordList, addPoissonNoise=True)
+
+        schema = SourceTable.makeMinimalSchema()
+
+        config = DetectAndMeasureTask.ConfigClass()
+        task = DetectAndMeasureTask(config=config, schema=schema)
+
+        butler = Butler(root=InputDir)
+        dataRef = butler.dataRef("calexp", dataId=dict(visit=1))
+        wcs = dataRef.get("raw").getWcs()
+        exposure.setWcs(wcs)
+        exposureIdInfo = dataRef.get("expIdInfo")
+        taskRes = task.run(exposure=exposure, exposureIdInfo=exposureIdInfo)
+        self.assertEqual(len(taskRes.sourceCat), numX * numY)
+        schema = taskRes.sourceCat.schema
+        centroidFlagKey = schema.find("slot_Centroid_flag").getKey()
+        parentKey = schema.find("parent").getKey()
+        psfFluxFlagKey = schema.find("slot_PsfFlux_flag").getKey()
+        psfFluxKey = schema.find("slot_PsfFlux_flux").getKey()
+        for src in taskRes.sourceCat:
+            self.assertFalse(src.get(centroidFlagKey))  # centroid found
+            self.assertEqual(src.get(parentKey), 0)     # not debelended
+            self.assertFalse(src.get(psfFluxFlagKey))   # flux measured
+            self.assertGreater(src.get(psfFluxKey), 4000)  # flux sane
+
+    def makeCoordList(self, bbox, numX, numY, minCounts, maxCounts, sigma):
+        """Make a coordList for plantSources
+
+        Coords are uniformly spaced in a rectangular grid, with linearly increasing counts
+        """
+        dX = bbox.getWidth() / float(numX)
+        dY = bbox.getHeight() / float(numY)
+        minX = bbox.getMinX() + (dX / 2.0)
+        minY = bbox.getMinY() + (dY / 2.0)
+        dCounts = (maxCounts - minCounts) / (numX * numY - 1)
+    
+        coordList = []
+        counts = minCounts
+        for i in range(numX):
+            x = minX + (dX * i)
```

Comment:
Remove parens, no space either side of `*`.

I have a hard time thinking REO wrote this....